### PR TITLE
Create site Nav bar

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do?

* Create navbar

#### Description of Task to be completed?
- Create navbar component
- Add relevant images
- Add default css
- Add router file for links

#### How should this be manually tested?

* Clone and run this branch

#### Screenshots

Not logged in
* 
normal-user

Logged in sign out dropdown
* 
image

* 
image

### CODE REVIEW

File: src/components/AdminDashboard/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+      <div>
+        <NavBar />
+        
+        <div className="Disposable">
+          <div><ClusterResources /></div>
+        </div>
+        {/* <Footer /> */}
+      </div>
+    )
```
Comment (Mubangizi) [inline src/components/AdminDashboard/index.js]:
```suggestion
    );
```

File: src/components/App/index.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 
 /* These below are for showing purposes */
 /* THIS IS GOING TO BE REMOVED IN THE DASHBOARD FEATURE */
 
 function App() {
   return <div>
+    <NavBar />
     <div className="Disposable">
       <div><ClusterResources /></div>
     </div>
```
Comment (Mubangizi) [inline src/components/App/index.js]:
```suggestion
    </div>;
```

File: src/components/NavBar/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    "name": "Catalysts",
+    "resourceNumber": 0,
+    "id": 6
+  },
+  {
+    "name": "Backup",
+    "resourceNumber": 2,
+    "id": 7
+  }
+]
```
Comment (Mubangizi) [inline src/components/NavBar/index.js]:
```suggestion
];
```

File: src/components/NavBar/index.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          <NavLogo />
+        </div>
+        <div class="Nav-Btn">
+          <label for="Nav-Check">
+            <span></span>
+            <span></span>
+            <span></span>
+          </label>
+        </div>
+        <div class="Nav-Links">
```
Comment (Mubangizi) [inline src/components/NavBar/index.js]:
```suggestion
        <div className="Nav-Links">
```

Comment (Mubangizi) [inline src/components/NavBar/index.js]:
```suggestion
        <div className="Nav-Links">
```

File: src/index.js [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
+// import App from './components/App';
 import * as serviceWorker from './serviceWorker';
 import store from './redux/store';
 import { saveState } from './helpers/localStorage';
+import Routes from './Router';
+
 
 // update localstorage whenever state changes
 store.subscribe(() => {
     saveState(store.getState());
```
Comment (Mubangizi) [inline src/index.js]: indenting

Comment (neelxie) [reply src/index.js]: Could you kindly elaborate on what you mean with this comment?

**Comment by neelxie** [comment]
> My own suggestion i think the nav bar needs to reduce in height seems too big

@Mubangizi I have reduced it to around 70px. Kindly do checkout the changes.

File: src/components/NavBar/index.js (hunk 4 above)
Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi is this the same issue raised below?

Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi is this the same issue raised above?

File: src/components/App/index.js [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
 
 function App() {
   return <div>
+    <NavBar />
     <div className="Disposable">
       <div><ClusterResources /></div>
-    </div>
+    </div>;
     <Footer />
   </div>
```
Comment (Mubangizi) [inline src/components/App/index.js]: Colon should be here
```suggestion
  </div>;
```

File: src/components/App/index.js [hunk 7]
```diff
@@ -1,17 +1,18 @@
...
 /* These below are for showing purposes */
 /* THIS IS GOING TO BE REMOVED IN THE DASHBOARD FEATURE */
 
 function App() {
   return <div>
+    <NavBar />
     <div className="Disposable">
       <div><ClusterResources /></div>
-    </div>
+    </div>;
```
Comment (Mubangizi) [inline src/components/App/index.js]: prematurely terminates returns contents

```suggestion
    </div>
```

File: src/components/NavBar/index.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  // const toggle = () => setIsOpen(!isOpen);
+
+  return (
+    <React.Fragment>
+      <div className="Navigation">
+        <input type="checkbox" id="nav-check" />
+        <div className="NavContainer">
+          <NavLogo />
+        </div>
+        <div class="Nav-Btn">
```
Comment (Mubangizi) [inline src/components/NavBar/index.js]: Pascal Case please
```suggestion
        <div class="NavBtn">
```

**Comment by neelxie** [comment]
> Please consider having less files changed on your next PRs...
> Screenshot from 2020-02-12 15-10-25
> 14 files is a lot to review.
> Otherwise, good work Derek. 👏🏼

@StephenAraka you could save time by just running it locally and seeing the feature in "action".

File: src/components/NavBar/index.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+            <span></span>
+          </label>
+        </div>
+        <div className="Nav-Links">
+          {/* <Link to="/pricing">PRICING</Link>
+          <Link to="/docs">DOCS</Link> */}
+          <div className="Logged-In">
+            
+            <div className="Nav-dropdown Drop Profile">
+              <div className="Dropdown-Icon">
```
Comment (Mubangizi) [inline src/components/NavBar/index.js]: line 59, 65, 68, 71 and 70 need Pascal Case thank you

File: src/components/App/index.js (hunk 7 above)
Comment (neelxie) [reply src/components/App/index.js]: > prematurely terminates returns contents

@Mubangizi kindly do note that your comment here does not make grammatical sense, could you be more elaborate?

Comment (Mubangizi) [reply src/components/App/index.js]: Content of return statement were terminated prematurily

**Comment by neelxie** [comment]
> So far so good!
> A few changes to be made tho.
> Revisit this design and update the style of the two states of the secondary button.
> Thanks
> Screenshot 2020-02-13 at 10 59 25

@ckwagaba kindly do check out the changes.

File: src/components/AdminDashboard/index.js (hunk 1 above)
Comment (neelxie) [reply src/components/AdminDashboard/index.js]: @Mubangizi this has been fixed.

File: src/components/NavBar/index.js (hunk 3 above)
Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi this has been fixed.

File: src/components/NavBar/index.js (hunk 4 above)
Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi this has been fixed.

File: src/index.js (hunk 5 above)
Comment (neelxie) [reply src/index.js]: @Mubangizi this has been fixed.

File: src/components/App/index.js (hunk 6 above)
Comment (neelxie) [reply src/components/App/index.js]: @Mubangizi this has been fixed.

File: src/components/App/index.js (hunk 7 above)
Comment (neelxie) [reply src/components/App/index.js]: @Mubangizi this has been fixed.

File: src/components/NavBar/index.js (hunk 8 above)
Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi this has been fixed.

File: src/components/NavBar/index.js (hunk 9 above)
Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi this has been fixed.

**Comment by neelxie** [comment]
> The secondary button maintains the white border and text color on hover.
> Also, the font-weight for both buttons is `bold`.
> Please update those. Thanks
> Screenshot 2020-02-13 at 11 39 22

@ckwagaba this too has been fixed.

File: src/components/NavBar/index.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+            <div className="UserProfileName">
+              <div className="NameUser"><strong>Colin Wagaba</strong></div>
+            </div>
+            <div className="Nav-dropdown Drop Profile">
+
+              <div className="DropdownIcon">
+                <i class="arrow down"></i>
+              </div>
+              <div className="Dropdown-Content">
+                <Link to="/logout">Sign Out</Link>
```
Comment (Mubangizi) [inline src/components/NavBar/index.js]: Line 27, 30, 37 and 42 please enforce Pascal case

Comment (neelxie) [reply src/components/NavBar/index.js]: @Mubangizi kindly do re-check this out.

File: src/components/NavBar/index.jsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+          {/* <Link to="/pricing">PRICING</Link>
+          <Link to="/docs">DOCS</Link> */}
+          <div className="LoggedIn">
+            <div className="DropdownIcon Showing">
+              <img src={PROFILE} size="1x" alt="profile" />
+            </div>
+            <div className="UserProfileName">
+              <div className="NameUser"><strong>Colin Wagaba</strong></div>
+            </div>
+            <div className="Nav-dropdown Drop Profile">
```
Comment (Mubangizi) [inline src/components/NavBar/index.jsx]: Pascal Case please
```suggestion
            <div className="NavDropdown Drop Profile">
```

File: src/components/NavBar/index.jsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+          <NavLogo />
+        </div>
+        {/* <div className="Nav-Btn">
+          <label htmlFor="Nav-Check">
+            <span />
+            <span />
+            <span />
+          </label>
+        </div> */}
+        <div className="NavLinks">
```
Comment (Mubangizi) [inline src/components/NavBar/index.jsx]: Some of the `NavLinks`  references were not refactored

File: src/components/NavBar/NavBar.css [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+  }
+  .Navigation > .Nav-links > Link {
+    display: block;
+    width: 100%;
+  }
+  .Navigation > #nav-check:not(:checked) ~ .Nav-links {
+    height: 0px;
+  }
+  .Navigation > #nav-check:checked ~ .Nav-links {
+    height: calc(100vh - 50px);
```
Comment (Mubangizi) [inline src/components/NavBar/NavBar.css]: `Nav-links` and `Nav-btn` these are not called anywhere

Comment (neelxie) [reply src/components/NavBar/NavBar.css]: @Mubangizi this has been fixed

File: src/components/NavBar/NavBar.css [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+/* .Navigation > .Nav-btn {
+  display: none;
+} */
+
+.Navigation > .NavLinks {
+  align-self: flex-end;
+  font-size: 18px;
+  height: 100%;
```
Comment (Mubangizi) [inline src/components/NavBar/NavBar.css]: Please Remove this line that is when the nav links become Centered Horizontally
```suggestion

```
This..
Screenshot from 2020-02-14 09-35-21

Not this
Screenshot from 2020-02-14 09-36-59

**Pull request merged**